### PR TITLE
Fix regression with the checkbox editor widget whereas clicking outside of the switch wouldn't change value

### PR DESCRIPTION
**Pull request by nirvn**
Discovered while debugging another issue.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 01b62700d17d02113f8a50dd5bf1f14b1511e0ab: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3687-da85c6c88899b3d5c28c6fabc123bb0888679bb4-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3687-da85c6c88899b3d5c28c6fabc123bb0888679bb4-arm-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3687-da85c6c88899b3d5c28c6fabc123bb0888679bb4-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3687-da85c6c88899b3d5c28c6fabc123bb0888679bb4-x86-android.apk)
- [**all access arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-3687-da85c6c88899b3d5c28c6fabc123bb0888679bb4-arm64-android.apk)
- [all access arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-3687-da85c6c88899b3d5c28c6fabc123bb0888679bb4-arm-android.apk)
- [all access x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-3687-da85c6c88899b3d5c28c6fabc123bb0888679bb4-x64-android.apk)
- [all access x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-3687-da85c6c88899b3d5c28c6fabc123bb0888679bb4-x86-android.apk)
</details>

**Review by suricactus — APPROVED** [review]
code looks good.

File: src/qml/editorwidgets/CheckBox.qml [hunk 1]
```diff
@@ -8,6 +8,9 @@ import "."
 EditorWidgetBase {
   height: childrenRect.height
 
+  //if the field type is boolean, ignore the configured 'CheckedState' and 'UncheckedState' values and work with true/false always
+  readonly property bool isBool: field.type == 1 //needs type coercion
```
Comment (suricactus) [inline src/qml/editorwidgets/CheckBox.qml:12]:
```suggestion
  readonly property bool isBool: field.type == 1 // needs type coercion
```

File: src/qml/editorwidgets/CheckBox.qml [hunk 2]
```diff
@@ -8,6 +8,9 @@ import "."
 EditorWidgetBase {
   height: childrenRect.height
 
+  //if the field type is boolean, ignore the configured 'CheckedState' and 'UncheckedState' values and work with true/false always
```
Comment (suricactus) [inline src/qml/editorwidgets/CheckBox.qml:11]:
```suggestion
  // if the field type is boolean, ignore the configured 'CheckedState' and 'UncheckedState' values and work with true/false always
```

**Pull request merged**